ansible_rulebook/websocket.py

```diff
@@ -58,68 +85,55 @@ async def _connect_websocket(
     while True:
-        backoff_delay = BACKOFF_MIN
         try:
             async with websockets.connect(
                 settings.websocket_url,
                 ssl=_sslcontext(),
                 extra_headers=extra_headers,
             ) as websocket:
+                # Connection succeeded - reset backoff delay
+                backoff_delay = BACKOFF_MIN
                 result = await handler(websocket, **kwargs)
                 if not retry:
                     break
         except asyncio.CancelledError:  # pragma: no cover
             raise
         except ShutdownException:
             break
-        except Exception as e:
-            status403_legacy = (
-                isinstance(e, websockets.exceptions.InvalidStatusCode)
-                and e.status_code == 403
-            )
-            status403 = (
-                isinstance(e, websockets.exceptions.InvalidStatus)
-                and e.response.status_code == 403
-            )
-            if status403_legacy or status403:
-                if refresh and settings.websocket_refresh_token:
-                    new_token = await renew_token()
-                    extra_headers["Authorization"] = f"Bearer {new_token}"
-                    # Only attempt to refresh token once. If a new token cannot
-                    # establish the connection, something else must cause 403
-                    refresh = False
-                else:
-                    raise
-            elif isinstance(e, OSError) and "[Errno 61]" in str(e):
-                # Sleep and retry implemention duplicated from
-                # websockets.lagacy.client.Client
-
-                # Add a random initial delay between 0 and 5 seconds.
-                # See 7.2.3. Recovering from Abnormal Closure in RFC 6544.
-                if backoff_delay == BACKOFF_MIN:
-                    initial_delay = random.random() * BACKOFF_INITIAL
-                    logger.info(
-                        "! connect failed; reconnecting in %.1f seconds",
-                        initial_delay,
-                        exc_info=True,
-                    )
-                    await asyncio.sleep(initial_delay)
-                else:
-                    logger.info(
-                        "! connect failed again; retrying in %d seconds",
-                        int(backoff_delay),
-                        exc_info=True,
-                    )
-                    await asyncio.sleep(int(backoff_delay))
-                # Increase delay with truncated exponential backoff.
-                backoff_delay = backoff_delay * BACKOFF_FACTOR
-                backoff_delay = min(backoff_delay, BACKOFF_MAX)
-                continue
-        else:
-            # Connection succeeded - reset backoff delay
-            backoff_delay = BACKOFF_MIN
-            refresh = True
+        except websockets.exceptions.InvalidStatusCode as e:
+            if refresh and e.status_code == 403:
```

Comment:
@bzwei Can the 2 403's logic be put into a function, they are running the same code